File: 1523 Count Odd Numbers in an Interval Range/1523count-odd-numbers-in-an-interval-range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        counts = high - low + 1

        if low % 2 == 0 and high % 2 == 0:
            return (counts - 1) // 2
        elif (low % 2 == 0 and high % 2 != 0) or (low % 2 != 0 and high % 2 ==0):
            return counts // 2
        else:
            return (counts - 1) // 2 + 1  
        

File: 1523 Count Odd Numbers in an Interval Range/1523count-odd-numbers-in-an-interval-range.cpp
class Solution {
public:
    int countOdds(int low, int high) {
        bool isLowOdd = (low % 2) != 0;
        bool isHighOdd = (high % 2) != 0;

        if (low == high && isLowOdd){
            return 1;
        } else if (isLowOdd && isHighOdd){
            return 2 + (high - low - 1) / 2;
        } else if (!isLowOdd && !isHighOdd){
            return std::ceil((high - low - 1) / 2.0);
        } else {
            return (high - low + 1) / 2;
        } 
    }
};